### PR TITLE
JBPM-6600: Stunner - Add BPMN Start Message Event

### PR DESCRIPTION
**Pull request by alepintus**
Hi @romartin  @hasys ,
Start Message Event added.

Could you please take a look?

Thanks!

fyi @wmedvede

### CODE REVIEW

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/legacy/Bpmn2JsonMarshaller.java [hunk 1]
```diff
@@ -262,14 +262,59 @@
...
-
+    private static final Logger _logger = LoggerFactory.getLogger(Bpmn2JsonMarshaller.class);
     private Map<String, DiagramElement> _diagramElements = new HashMap<String, DiagramElement>();
     private Map<String, Association> _diagramAssociations = new HashMap<String, Association>();
     private Scenario _simulationScenario = null;
-    private static final Logger _logger = LoggerFactory.getLogger(Bpmn2JsonMarshaller.class);
     private IDiagramProfile profile;
     private boolean coordianteManipulation = true;
 
+    private static String unescapeXML(String str) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/legacy/Bpmn2JsonMarshaller.java]: Hey  @alepintus , I understand this part is not new... you just moved code around, right? I mean not sure if applies but trying to un/escape xml can be done easier and in a safe way for example using the Apache Commons libraries (as this is backend code np), for example` StringEscapeUtils#escapeXml`, etc, but tbh not really sure if that applies here...

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/legacy/Bpmn2JsonUnmarshaller.java [hunk 2]
```diff
@@ -246,6 +239,46 @@ public Bpmn2JsonUnmarshaller() {
         }
     }
 
+    private static String escapeXmlString(String string) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/legacy/Bpmn2JsonUnmarshaller.java]: Same as comment above, we could use Apache Commons libraries here... if applies.

**Comment by romartin** [comment]
ok to test

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by wmedvede** [comment]
Hello Alessio, I believe that for this Message event we should apply the same strategy I used for the Error events (and that I'll apply soon for the signal events). We should ideally collect the MessageRef form MessageRefs already defined in the process or let the user add a new one if desired.
For this you'll have to create e.g. ProcessMessageRefProvider similar to the one in the following link.

https://github.com/wmedvede/kie-wb-common/blob/61ccb56fa134ab175e105a2d649f71eff0e5a834/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/dataproviders/ProcessErrorRefProvider.java#L36-L36

see here how the provider is used

https://github.com/wmedvede/kie-wb-common/blob/61ccb56fa134ab175e105a2d649f71eff0e5a834/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/event/error/ErrorEventExecutionSet.java#L46-L46

**Comment by wmedvede** [comment]
Hello @alepintus , I'll also suggest that with a strategy similar like the one @romartin introduced in his last PR for the case of the IntermediateSignalEventCatching, you also add the following effect:

    "when the user clicks the isInterrupting" attribute we should make the border of the icon solid or dashed depending on the attribute value. If isInterrupting == false the border should be dashed, and if isInterrupting == true then the border should be solid." 

(this is something that I'll also have to review on my events, since it's a new feature in the graphics/view/shapes side) 

see here:

https://github.com/wmedvede/kie-wb-common/blob/7b911b05a772db1aca8622e374a1fcf1dde86b85/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/shape/view/handler/EventCancelActivityViewHandler.java#L26

I believe that in the case of the isInterrupting attribute we can probably create a EventIsInterruptingViewHandler for the BaseStartEvent class, and proceed very similar.

**Comment by alepintus** [comment]
Hi @romartin , @wmedvede , @hasys , 
code changed as suggested to un/escape xml with Apache Commons libraries, 
Added new ProcessMessageRefProvider and now when the user clicks isInterrupting attribute the border of the StartMessageEvent becomes solid or dashed depending on the attribute value.

Could you please take a look?

Thanks!

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
ok to test

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
@hasys @wmedvede missing your review/approval here guys

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/resources/org/kie/workbench/common/stunner/bpmn/client/resources/images/icons/event/event-start.svg [hunk 3]
```diff
@@ -23,4 +23,7 @@
 	<g transform="translate(4,4) scale(0.05,0.05)" style="opacity:1">
 		<use xlink:href="event-timer.svg#eventTimer"/>
 	</g>
+	<g stunner:layout="CENTER" transform="translate(4,4) scale(0.05,0.05)" style="opacity:1">
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/resources/org/kie/workbench/common/stunner/bpmn/client/resources/images/icons/event/event-start.svg]: hey @alepintus ! Just realized here was an error that was causing some icons not being displayed. No problem, was an easy fix, but just let you know that the error was that you added a new attribute `stunner:layout` but you didn't add the `stunner` namespace declaration on top. 
Anyway np, just fixed it here https://github.com/kiegroup/kie-wb-common/pull/1328 In fact that `stunner:layout` for now is not necessary here, so I just removed. 
Thanks!

**Pull request merged**